lib/pipelineFactory.js

```diff
@@ -42,8 +42,22 @@ class PipelineFactory extends BaseFactory {
         const modelConfig = hoek.applyToDefaults({
             createTime: (new Date()).toISOString()
         }, config);
+        const admin = config.admins[0];
 
-        return super.create(modelConfig);
+        return Pipeline.getScmRepo(admin, this.scmPlugin, config.scmUrl)
```

Comment:
That's reasonable. I have moved `getScmRepo` to the pipelineFactory. I think `getRepoData` in the scmPlugin is not necessary in this case.